### DISCUSSION TITLE
Invoke-PnPSPRestMethod to download the contents of a thumbnail image file

### DISCUSSION THREAD

**Discussion by shsharm85** [pnp/powershell#1709] [Q&A]
I am using Invoke-PnPSPRestMethod to download the contents of a thumbnail image file (for SPO News articles). SharePoint looks to be storing the image in a link similar to below (shortened for simplicity)

$url1 = [https://tenant.sharepoint.com/_api/v2.1/shares/.../driveItem/thumbnails/0/c400x99999/content?preferNoRedirect=true](https://tenant.sharepoint.com/_api/v2.1/shares/.../driveItem/thumbnails/0/c400x99999/content?preferNoRedirect=true&xsdata=MDN8MDF8fGViMjgyYTU3NmQxZTRjNDc4ZTAyYzNiMWMyZGExZjlkfGUyOTQyMGY3ZjBiNjRiZGZhOGMzYTRlYWNmZjg2ZTQwfDB8MHw2Mzc4MzUwOTIzNDcyOTcwNDd8R29vZHxWR1ZoYlhOVFpXTjFjbWwwZVZObGNuWnBZMlY4ZXlKV0lqb2lNQzR3TGpBd01EQWlMQ0pRSWpvaVYybHVNeklpTENKQlRpSTZJazkwYUdWeUlpd2lWMVFpT2pFeGZRPT0%3D&sdata=dGcyemlMU0Q2Y2xiZW1WcXFaTHdha1RNMkdSMWdqVmlIL0Jiand4bnBnMD0%3D&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2Cshsharm%40microsoft.com)"

Using the following
$output = Invoke-PnPSPRestMethod -url1 $picFile -ContentType 'application/octet-stream'

I get the following error.
Message : {"error":{"code":"notAcceptable","message":"Accept header cannot be satisfied (Accept: 'application/json; odata=nometadata', Content-Type: 'image/png')"}}

I am able to retrieve content with the following (removed the redirect at end of the link)

$url2 = [https://tenant.sharepoint.com/_api/v2.1/shares/.../driveItem/thumbnails/0/c400x99999/content](https://tenant.sharepoint.com/_api/v2.1/shares/.../driveItem/thumbnails/0/c400x99999/content?xsdata=MDN8MDF8fGViMjgyYTU3NmQxZTRjNDc4ZTAyYzNiMWMyZGExZjlkfGUyOTQyMGY3ZjBiNjRiZGZhOGMzYTRlYWNmZjg2ZTQwfDB8MHw2Mzc4MzUwOTIzNDcyOTcwNDd8R29vZHxWR1ZoYlhOVFpXTjFjbWwwZVZObGNuWnBZMlY4ZXlKV0lqb2lNQzR3TGpBd01EQWlMQ0pRSWpvaVYybHVNeklpTENKQlRpSTZJazkwYUdWeUlpd2lWMVFpT2pFeGZRPT0%3D&sdata=dGcyemlMU0Q2Y2xiZW1WcXFaTHdha1RNMkdSMWdqVmlIL0Jiand4bnBnMD0%3D&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2Cshsharm%40microsoft.com)"
$output = Invoke-PnPSPRestMethod -url $url2 -Raw
$output | Out-File -FilePath C:\temp\pic.png

In this case the pic.png is not recognizable as image (is 125KB, vs the actual image is 49 KB).

Any ideas?

**Comment by engineererr** [comment]
I download images from the search result property 'BannerImageUrlOWSURLH'.
The same method should work for other SPO endpoints.

```
$res = Submit-PnPSearchQuery -Query 'news' -SelectProperties BannerImageUrlOWSURLH
$row = $res.ResultRows[0]
$pictureUrlRaw = $row["BannerImageUrlOWSURLH"]
$pictureUrl = $pictureUrlRaw.Split(",")[0].Trim() + "&resolution=5"
$accessToken = Get-PnPAccessToken -ResourceTypeName SharePoint
Invoke-RestMethod -Method Get -Uri $pictureUrl -Headers @{Authorization = "Bearer $accessToken"} -OutFile C:\temp\pic.jpg
```
